accounts/abi/bind/bind.go

```diff
@@ -601,3 +629,36 @@ func hasStruct(t abi.Type) bool {
 		return false
 	}
 }
+
+func createPrecompileDataAndTemplate(contract *tmplContract, structs map[string]*tmplStruct) (interface{}, string) {
+	funcs := make(map[string]*tmplMethod)
+
+	for k, v := range contract.Transacts {
+		funcs[k] = v
+	}
+
+	for k, v := range contract.Calls {
+		funcs[k] = v
+	}
+
+	_, isAllowList := funcs[readAllowListFuncKey]
```

Comment:
I wonder if this may unintentionally interfere with anyone else's contracts and if there's another way to do this. Is it possible to parse the contract to see if it is something like:

```
interface NameOfContract is IAllowList
```

instead of checking for a specific function name?

Reply:
Generator parses ABI not the contract. it's kind-of a janky way to fetch this from ABI but I don't think ABI includes inheritance info. Parsing the contract would be a very different thing than the current approach. We can make it stricter with checking all of AllowList funcs (`readAllowList` + `setAdmin` + setEnabled` + `setNone`).

Reply:
I'd be in favor of making it stricter.

Either way, we should also document that overlapping with these function names will result in unintended behavior.